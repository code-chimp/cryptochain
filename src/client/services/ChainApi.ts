import IBlock from '../../@interfaces/IBlock';

export const fetchBlocks = async (): Promise<Array<IBlock>> => {
  try {
    const result = await fetch('/api/blocks');

    return (await result.json()) as Array<IBlock>;
  } catch (e) {
    throw new Error(`ChainApi.fetchBlocks: ${(e as { message?: string }).message}`);
  }
};
