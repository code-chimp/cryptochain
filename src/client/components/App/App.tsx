import React, { FC, useEffect, useState } from 'react';
import { fetchBlocks } from '../../services/ChainApi';

const App: FC = () => {
  const [count, setCount] = useState<number>(0);

  useEffect(() => {
    fetchBlocks().then(blocks => setCount(blocks.length));
  }, []);

  return (
    <div>
      <h1>Welcome to Cryptochain!</h1>
      <h3>[stub front-end functionality]</h3>
      <p>There are currently {count} block(s) on the chain</p>
    </div>
  );
};

export default App;
