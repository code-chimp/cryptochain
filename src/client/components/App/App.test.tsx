/** @jest-environment jsdom */
import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

import * as chainApi from '../../services/ChainApi';
jest.mock('../../services/ChainApi');

describe('components / App', () => {
  it('should match the snapshot', async () => {
    (chainApi.fetchBlocks as jest.Mock).mockResolvedValueOnce([]);

    const component = render(<App />);

    expect(component.container.firstChild).toMatchSnapshot();
  });
});
