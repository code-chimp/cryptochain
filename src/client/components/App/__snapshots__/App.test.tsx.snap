// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components / App should match the snapshot 1`] = `
<div>
  <h1>
    Welcome to Cryptochain!
  </h1>
  <h3>
    [stub front-end functionality]
  </h3>
  <p>
    There are currently 
    0
     block(s) on the chain
  </p>
</div>
`;
